docs/src/main/docs/security/05_extensibility.adoc

```diff
@@ -0,0 +1,129 @@
+To create a custom authentication provider, create a class that implements
+`io.helidon.security.spi.AuthenticationProvider`.
+Implementation is responsible for taking a request and asserting a subject
+based on that request.
+In case the protocol is multi-request (e.g. challenge for basic authentication),
+ you have the possibility to return specific headers and a response code. The
+ default semantics of these is HTTP, though providers may exist that are not
+ HTTP specific.
+
+=== Authorization provider
+To create a custom authorization provider, create a class that implements
+`io.helidon.security.spi.AuthorizationProvider`.
+Implementation is responsible for taking a request and checking whether the
+ request can continue processing (e.g. if the current user and/or service subject
+ has a right to execute it).
+
+If authentication is configured, the Security component guarantees it resolved
+ before authorization.
+
+=== Outbound security provider
+To create a custom outbound security provider, create a class that implements
+`io.helidon.security.spi.OutboundSecurityProvider`.
+Implementation can update outgoing message headers to handle
+security for an outgoing request (e.g. identity propagation, mapping etc.).
+
+=== Audit provider
+To create a custom audit provider, create a class that implements
+`io.helidon.security.spi.AuditProvider`.
+Security component feeds each audit provider all messages from all components
+that invoke audit method on "Security" class, including internal audit events
+pre-configured in the component itself (e.g. authentication, authorization
+ events).
+
+Implementation may do whatever desired with these messages, e.g.:
+
+* filter them
+* log them
+* store them to a database
+* forward them to an audit component
+* discard them
+
+== Provider selection policy
+Each request is processed by a single authentication and/or authorization
+ provider. The selection policy provides the security component information about
+ which provider to use. Out of the box, there are three policies:
+
+1. "First" policy - first configured provider (or explicitly defined default
+ provider) is used by default, if a named provider is requested, it would be used
+2. "Composite" policy - this policy allows for a sequence of providers to be
+ executed (e.g. one request may have more than one provider) - used for example to
+ resolve service and user authentication
+3. "Class" policy - this allows usage of a custom policy defined by fully
+ qualified class name
+
+To create a custom provider selection policy, create a class that implements
+"io.helidon.security.spi.ProviderSelectionPolicy".
+
+== Framework integration
+Security provides integration with Helidon WebServer (`helidon-security-integration-webserver`)
+ and with Jersey (`helidon-security-integration-jersey`).
+
+These integrations use APIs that are available to integrate any framework/application (for example
+we could integrate security with messaging, such as JMS).
+
```

Comment:
1. When you say "These integrations", did you mean that the `helidon-security-integration-webserver` and `helidon-security-integration-jersey` modules?

2. Does the phrase "use APIs that are available" refer to APIs that we provide via the helidon modules, or to Java APIs like JMS ?

Depending on the answers to these questions, we might need to rephrase this part.